.circleci/config.yml

```diff
@@ -88,14 +93,15 @@ jobs:
           at: ~/repo/
       - restore_cache:
           keys:
-            - v3-dependencies-{{ checksum "Pipfile.lock" }}
-      - run: *install-pipenv
-      - run: pip install aws-sam-cli
+            - v4-dependencies-{{ checksum "uv.lock" }}
+      - install_uv
+      - run: uv venv
+      - run: uv pip install aws-sam-cli
```

Comment:
Why do we need to install aws-sam-cli here if it is in our manifest?

I think we should either:
- Have aws-sam-cli pinned in our manifest and install it with `uv sync` (it might actually make sense to put it in its own optional group)
OR
- Install the latest version ad-hoc in the CI build

but not both. It just creates confusion if we're trying to debug something because the version we're using in CI then isn't the version pinned in the manifest

Reply:
I believe it's only in the manifest as part of the workaround that @symroe did (and I copied) in https://github.com/DemocracyClub/electionleaflets/commit/a2b7ea12a934c4240f22147871e7950f89db24cb . So I believe whenever the issue with SAM resolves we won't need aws-sam-cli in the manifest anymore? Maybe @symroe can confirm.

Reply:
I removed SAM CLI from the manifest and switched to using the `public.ecr.aws/sam/build-python3.12:latest` base image for the `sam_` builds. Same as we are doing on Leaflets